### DISCUSSION TITLE
How to add a conditional state transition

### DISCUSSION THREAD

**Discussion by vikasaroy** [spatie/laravel-model-states#199] [Q&A]
Hey Guys,

I am trying to implement this package in a hyperlocal delivery app. Some of the orders are Cash on Delivery(also known as Pay On Delivery).

I want to add an extra state transition when the order is a COD order. 

How can I achieve this using this package?

Any hint will be appreciated.

**Comment by brendt** [comment]
You'd have to manage the state flow manually with this package. There's no automated way to allow conditional transitions.

Transitions are allowed to prevent them being executed, but that will result in an exception, which is not what you're after.
